lib/util/mp4_generator.js

```diff
@@ -11,33 +11,43 @@ goog.require('shaka.util.ManifestParserUtils');
 goog.require('shaka.util.Uint8ArrayUtils');
 
 
-/**
- * @export
- */
 shaka.util.Mp4Generator = class {
   /**
-   * @param {shaka.extern.Stream} stream
-   * @param {number} timescale
-   * @param {?number} duration
-   * @param {?Array.<string>} videoNalus
+   * @param {shaka.util.Mp4Generator.StreamInfo} streamInfo
    */
-  constructor(stream, timescale, duration, videoNalus) {
+  constructor(streamInfo) {
     shaka.util.Mp4Generator.initStaticProperties_();
 
     /** @private {shaka.extern.Stream} */
-    this.stream_ = stream;
+    this.stream_ = streamInfo.stream;
```

Comment:
I feel like this should be marked as non-optional too (as well as the stream object inside the streamInfo).